src/libs/API.js

```diff
@@ -127,7 +128,14 @@ function handleExpiredAuthToken(originalCommand, originalParameters, originalTyp
             // Now that the API is authenticated, make the original request again with the new authToken
             const params = addDefaultValuesToParameters(originalCommand, originalParameters);
             return Network.post(originalCommand, params, originalType);
-        });
+        })
+        .catch(() => (
+
+            // If the request did not succeed and we were not logged out requeue the original request. We are using
```

Comment:
>we are in a case where we know the API was just authenticated

At this point in the code we got here because `reauthenticate()` threw an error

>how is it possible that Network.post() can fail with authentication

Because `fetch()` failed. Most likely you are offline when we tried to call `Authenticate` and so you got an `API is offline` error. But you otherwise have all the necessary credentials to re-authenticate.

>if it does fail because of authentication, why would it help to re-authenticate again

Ah sorry, I'm not too sure what "fail because of authentication" means here. So maybe you can clarify the question for me. But we are making a new `request()` here so that when you come back online you make a `request()` instead of a `Network.post()`. `Network.post()` doesn't attempt to re-authenticate.

I think it might help to summarize that context in a comment.

Reply:
> fail because of authentication

I'm speaking primarily of the case where the authToken has expired. I didn't think of the case for the API being offline, so I think this makes more sense.